### PR TITLE
Feature/offpay 254

### PR DESCRIPTION
**Pull request by habeeb01**
added part and parcel flow logic and tests

### CODE REVIEW

File: app/uk/gov/hmrc/offpayroll/models/PartAndParcelCluster.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  val clusterElements: List[Element] = List(
+    Element("workerReceivesBenefits", RADIO, 0, this),
+    Element("workerAsLineManager", RADIO, 1, this),
+    Element("contactWithEngagerCustomer", RADIO, 2, this),
+    Element("workerRepresentsEngagerBusiness", RADIO, 3, this)
+  )
+
+  private val flows = List(
+    FlowElement("partParcel.workerReceivesBenefits", Map("partParcel.workerReceivesBenefits" -> "No"), Option("partParcel.workerAsLineManager")),
```
Comment (peteanning) [inline app/uk/gov/hmrc/offpayroll/models/PartAndParcelCluster.scala]: Formatting long lines

File: test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  it should " ask for a decision when all questions have been asked " in {
+    val currentQnA = ("partParcel.workerRepresentsEngagerBusiness", "Yes")
+    val allAnswers = PropertyFileLoader.transformMapToAListOfAnswers(propsFilteredByCluster)
+
+    partAndParcelCluster.shouldAskForDecision(allAnswers, currentQnA).isEmpty shouldBe true
+  }
+
+  it should " ask the correct next question when there is some flow logic and there are no more questions " in {
```
Comment (peteanning) [inline test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala]: re-word to make sense

File: test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    val currentQnA = ("partParcel.workerReceivesBenefits", "Yes")
+    val partialAnswers = List(currentQnA)
+
+    partAndParcelCluster.shouldAskForDecision(partialAnswers, currentQnA).isEmpty shouldBe true
+  }
+
+  it should " ask the correct next question when there is some flow logic and there are no more questions - another flavour " in {
```
Comment (peteanning) [inline test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala]: re-word description

File: app/uk/gov/hmrc/offpayroll/models/PartAndParcelCluster.scala (hunk 1 above)
Comment (habeeb01) [inline app/uk/gov/hmrc/offpayroll/models/PartAndParcelCluster.scala]: done

File: test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala (hunk 2 above)
Comment (habeeb01) [inline test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala]: done

File: test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala (hunk 3 above)
Comment (habeeb01) [inline test/uk/gov/hmrc/offpayroll/PartAndParcelClusterSpec.scala]: done

**Pull request merged**